### ISSUE TITLE
Crash when students use Rosefire: Rosefire is not defined

### ISSUE THREAD

**Issue by fisherds** [Rosebotics/RoseboticsWeb#29]
Bug:
Student click the button to link their Rose-Hulman account and nothing happens.

Console error message:
In the console the error message says "Uncaught ReferenceError: Rosefire is not defined"

I didn't debug on the student computer, but the crash must come from here in rosebotics.js (only place Rosefire is used):
Rosefire.signIn(registryToken, function(err, token) { ...

Feels like that Rosefire.signin command doesn't exist.  I figured it must be an issue with a script tag.  Here are the two hits I found:

<script src="/static/assets/lib/rosefire.min.js"></script>

<script src="https://cdn.rawgit.com/rockwotj/rosefire/js-v1.0.4/dist/js/rosefire.min.js"></script>

Both work fine and really that script is pretty simple.  Rosefire is not defined there it just opens the new tab with the registry token info.  So who the heck exports Rosefire. :)  Seems like this script is just missing....

https://ada.csse.rose-hulman.edu/rosefire/gae-example/blob/master/static/js/rosefire.js

Funny that I know how to use Rosefire with AppEngine for my courses, but it looks like it's done differently in rosebotics.org

**Comment by fisherds** [comment]
BTW the "fix" for now is to have the affected students (not everyone has this bug) use my computer to resolve the issue.  Don't know what is special about their computer yet.

**Comment by fisherds** [comment]
So I realize that:
http://www.rosebotics.org/static/assets/lib/rosefire.min.js

is just a minified version of:
https://ada.csse.rose-hulman.edu/rosefire/gae-example/blob/master/static/js/rosefire.js

So Rosefire should exist on their computers.  It just uses a weird export style that is unusual to me (maybe that weird style is causing the issue).

Next tests:
On a student computer that has this issue try the following:
- See if you can type Rosefire in the console.  It should say something like:
```
{signIn: ƒ}
signIn: ƒ (e,n)
__proto__: Object
```
but I'm guessing it won't.
- Next try to manually visit http://www.rosebotics.org/static/assets/lib/rosefire.min.js from there computer to see if the script loads there
 - Next check to make sure that script did load using the Chrome dev tools from their computer
 - Next try a logout then login to see if that just magically fixes things up
- Next try a code fix.  Instead of the fancy approach used by that js file, just do it simple in rosefire.js with no minification no exports function stuff, just do it all as simple global variables on the window.  For example...

```
Rosefire = {};
Rosefire.signIn = function (registryToken, callback) {
    var token = encodeURIComponent(registryToken);
    var origin = encodeURIComponent(location.origin);
    var rosefireWindow = window.open('https://rosefire.csse.rose-hulman.edu/webview/login?platform=web&registryToken=' + token + '&referrer=' + origin, '_blank');
    // Hacky, but seems to be the only method.
    var intervalId = setInterval(function () {
        if (rosefireWindow.closed) {
            clearInterval(intervalId);
            if (callback) {
                callback(new Error('Login cancelled'));
                callback = null;
            }
        }
    }, 500);
    window.addEventListener('message', function (event) {
        var origin = event.origin || event.originalEvent.origin;
        if (origin !== 'https://rosefire.csse.rose-hulman.edu') {
            console.error('Invalid origin:' + origin);
            return;
        }
        var cb = callback;
        callback = null;
        clearInterval(intervalId);
        event.source.close();
        if (cb) {
            cb(null, parseJWT(event.data));
        }
    });
};


var parseJWT = function (jwt) {
    var payload = jwt.split(".")[1];
    payload = JSON.parse(atob(payload));
    if (!!payload.d) {
        // old format
        payload = payload.d;
    } else {
        // new format
        var username = payload.uid;
        payload = payload.claims;
        payload.uid = username;
    }
    return {
        token: jwt,
        name: payload.name,
        group: payload.group,
        email: payload.email,
        username: payload.uid
    }
};
```

**Comment by fisherds** [comment]
Having the student log out and log back in did fix the issue for one student.

**Comment by fisherds** [comment]
https://github.com/Rosebotics/RoseboticsWeb/commit/49fed472789875b187dcabc790c2714cdf555f9c

Included Rosefire on CourseBuilder pages .
After a short talk with Tyler we found the bug.

Indeed all CourseBuilder pages had a button for edit profile but no
Rosefire.  If a student only ever visited CourseBuilder pages they
couldn’t link their Rose-Hulman account.  Added the Rosefire script to
those pages as well by adding it to the CB base.html file.
